src/java/htsjdk/samtools/SamFileValidator.java

```diff
@@ -90,7 +90,7 @@
     private Set<Type> errorsToIgnore = EnumSet.noneOf(Type.class);
     private boolean ignoreWarnings = false;
     private boolean bisulfiteSequenced = false;
-    private boolean validateIndex = false;
+    private ValidationStringency indexValidationStringency = ValidationStringency.SILENT;
```

Comment:
Two things:
1. I don't love the use of `ValidationStringency` here for controlling how much index validation is done, since the values tend to imply how errors are _reported_, not how much validation is applied.  I won't object to leaving it this way, but it might be better to either have a boolean that controls whether validation is exhaustive or less exhaustive
2. The default should not be SILENT, since this changes default behaviour very significantly from exhaustive validation to no validation.  I'd be ok with changing the default to less exhaustive, but not omitting validation entirely.